### ISSUE TITLE
Error object differs between pg and pg-native

### ISSUE THREAD

**Issue by deyhle** [brianc/node-postgres#938]
> node-postgres abstracts over the pg-native module to provide exactly the same interface as the pure JavaScript version. No other code changes are required. If you find yourself having to change code other than the require statement when switching from require('pg') to require('pg').native please report an issue.

The fix for #743 made me test a switch to pg-native again. However, I still found an API difference: the Error object. While the default pg library returns an error with properties `code` and `detail`, the parameters are named differently when using pg-native.

`require('pg')`:

``` js
{
  "name": "error",
  "length": 199,
  "severity": "ERROR",
  "code": "23505",
  "detail": "Key (c1)=(one) already exists.",
  "schema": "public",
  "table": "dbhandlertest",
  "constraint": "dbhandlertest_pkey",
  "file": "nbtinsert.c",
  "line": "402",
  "routine": "_bt_check_unique"
}
```

`require('pg').native`:

``` js
{
  "severity": "ERROR",
  "sqlState": "23505",
  "messagePrimary": "duplicate key value violates unique constraint \"dbhandlertest_pkey\"",
  "messageDetail": "Key (c1)=(one) already exists.",
  "sourceFile": "nbtinsert.c",
  "sourceLine": "402",
  "sourceFunction": "_bt_check_unique"
}
```

This makes it hard to do proper error handling. Especially the `code` property is probably used a lot for that.

**Comment by abenhamdine** [comment]
related to #919

**Comment by brianc** [comment]
Adding to the `pg@7.0` release milestone to normalize these fields across the native bindings & the pure javascript library.

**Comment by vitaly-t** [comment]
> This makes it hard to do proper error handling. Especially the code property is probably used a lot for that.

In the meantime, the generic error handler can simply accept the `native` flag and parse errors differently.

**Comment by brianc** [comment]
I have this fixed on the 7.0 branch.
